tests/components/gree/test_climate.py

```diff
@@ -843,6 +870,40 @@ async def test_update_swing_mode(
     assert state.attributes.get(ATTR_SWING_MODE) == swing_mode
 
 
+async def test_coordinator_update_handler(
+    hass: HomeAssistant, discovery, device
+) -> None:
+    """Test for coordinator update handler."""
+    await async_setup_gree(hass)
+    await hass.async_block_till_done()
+
+    entity: GreeClimateEntity = hass.data[DOMAIN].get_entity(ENTITY_ID)
```

Comment:
Please don't interact with the entity instance directly in the tests. Only use the core state from the state machine.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations